explore/src/main/scala/explore/config/ConfigurationPanel.scala

```diff
@@ -46,62 +51,89 @@ import sttp.model._
 import scala.concurrent.duration._
 
 final case class ConfigurationPanel(
-  id:               Option[Observation.Id],
-  renderInTitle:    Tile.RenderInTitle
-)(implicit val ctx: AppContextIO)
+  obsId:               Observation.Id,
+  scienceRequirements: View[ScienceRequirementsData],
+  undoStacks:          View[UndoStacks[IO, ScienceRequirementsData]],
+  renderInTitle:       Tile.RenderInTitle
+)(implicit val ctx:    AppContextIO)
     extends ReactProps[ConfigurationPanel](ConfigurationPanel.component)
 
 object ConfigurationPanel {
   type Props = ConfigurationPanel
 
-  implicit val modeDisplay: Display[ConfigurationMode]             = Display.by(_.label, _.label)
+  implicit val modeDisplay: Display[ScienceMode] = Display.byShortName {
+    case ScienceMode.Spectroscopy => "Spectroscopy"
+    case ScienceMode.Imaging      => "Imaging"
+  }
+
   implicit val specCapabDisplay: Display[SpectroscopyCapabilities] = Display.by(_.label, _.label)
   implicit val focaLPlaneDisplay: Display[FocalPlane]              = Display.by(_.label, _.label)
   implicit val propsReuse: Reusability[Props]                      = Reusability.derive
   implicit val stateReuse: Reusability[State]                      = Reusability.never
 
   @Lenses
   final case class State(
-    mode:                ConfigurationMode,
-    spectroscopyOptions: SpectroscopyConfigurationOptions,
-    imagingOptions:      ImagingConfigurationOptions,
-    matrix:              Pot[SpectroscopyModesMatrix]
+    mode:           ScienceMode,
+    imagingOptions: ImagingConfigurationOptions,
+    matrix:         Pot[SpectroscopyModesMatrix]
   )
 
-  case class UndoView(
-    undoCtx: UndoCtx[State]
-  ) {
-    private val undoableView = UndoableView(undoCtx)
-
-    def apply[A](
-      modelGet: State => A,
-      modelMod: (A => A) => State => State
-    ): View[A] =
-      undoableView.apply(
-        modelGet,
-        modelMod,
-        _ => IO.unit
-      )
-
-    def apply[A](
-      lens: Lens[State, A]
-    ): View[A] =
-      apply(lens.get, lens.modify)
-
-  }
+  val dataIso: Iso[SpectroscopyRequirementsData, SpectroscopyConfigurationOptions] =
+    Iso[SpectroscopyRequirementsData, SpectroscopyConfigurationOptions] { s =>
+      def wavelengthToMicro(w: Wavelength) = w.micrometer.toValue[BigDecimal]
```

Comment:
Can we use an `Iso[Wavelength, Quantity[BigDecimal, Micrometer]]` ?

Reply:
You can't do that. `Wavelength` of 0 is illegal while `Quantity` of 0 is

Reply:
I see, I missed that. It would have to be a `Prism` then. Or we could leave it as it is.